### ISSUE TITLE
Cannot intercept/prevent default event using (click)

### ISSUE THREAD

**Issue by gdalam** [ngx-toggle/ngx-toggle#26]
### Bug description:

Not sure if this would be classified as a bug or feature request, but I'm unable to use (click) and preventDefault/stopPropagation as you can with normal UI elements. The function fires, but the toggle is already complete by that point. This would be useful for validation purposes or taking other actions as part of the toggle.

### Link to minimally-working StackBlitz that reproduces the issue:

https://stackblitz.com/edit/angular-gp2fyk-huufee

### Versions of Angular, ng-bootstrap and Bootstrap:

Angular: 9.1.1

ng-bootstrap: 6.0.2

**Comment by trickeyone** [comment]
@gdalam When you use `(click)` that's Angular's basic click action. I've never looked into handling the root click action, nor is it triggered from inside the component. What are you looking for, functionality-wise, to happen if you call `event.preventDefault()`? That usually means something like the basic HTML element's default isn't triggered, i.e clicking on an anchor-link and it not changing the browser's location/sending you to a new URL.

**Comment by gdalam** [comment]
@trickeyone I was hoping to find a way to prevent the toggle from being flipped on click (without disabling), or more generally validate the action before the toggle is flipped.

In my current case I ended up having to actually disable the element, which works, I just wasn't super sold on that idea because of how it might be interpreted from an end-user/design standpoint. Nothing breaking, just might be some useful functionality.

As a side note, I tried setting the value of the model using `(click)` and what the rendered element displays gets out of sync with the actual model: https://stackblitz.com/edit/angular-gp2fyk-9gwzgn
